plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/N4JSScopeProvider.xtend

```diff
@@ -280,6 +287,13 @@ class N4JSScopeProvider extends AbstractScopeProvider implements IDelegatingScop
 	 */
 	private def IScope scope_IdentifierRef_id(IdentifierRef identifierRef, EReference ref) {
 		val VariableEnvironmentElement vee = identifierRef.ancestor(VariableEnvironmentElement);
+		val scope = getLexicalEnvironmentScope(vee, identifierRef, ref);
+		// Handle constructor visibility
+		if (identifierRef.eContainer instanceof NewExpression) {
```

Comment:
at first sight, this special case seems a bit odd to me, at this place. But maybe I just don't get it. Let's discuss later...